modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/service/test/DLFileEntryMetadataLocalServiceTest.java

```diff
@@ -253,6 +278,33 @@ public void testGetNoStructuresFileEntryMetadatas() throws Exception {
 		}
 	}
 
+	@Test
+	public void testUpdateFileEntryMetadataWithExternalReferenceCode()
+		throws Exception {
+
+		User user = TestPropsValues.getUser();
+
+		DLFileVersion dlFileVersion = _dlFileEntry.getFileVersion();
+
+		String externalReferenceCode = RandomTestUtil.randomString();
+
+		_dlFileEntryMetadataLocalService.updateFileEntryMetadata(
+			externalReferenceCode, _dlFileEntryType.getFileEntryTypeId(),
+			_dlFileEntry.getFileEntryId(), dlFileVersion.getFileVersionId(),
+			setUpDDMFormValuesMap(
+				_ddmStructure.getStructureKey(), user.getLocale()),
+			ServiceContextTestUtil.getServiceContext(_group.getGroupId()));
+
+		DLFileEntryMetadata fileEntryMetadata =
+			_dlFileEntryMetadataLocalService.getFileEntryMetadata(
+				_ddmStructure.getStructureId(),
+				dlFileVersion.getFileVersionId());
```

Comment:
Can you rename this to `dlFileEntryMetadata` for consistency with the other test method? (sorry, I didn't see this one in the previous review).

Reply:
no problem!

Thanks for reviewing :)